network/netplay/netplay.c

```diff
@@ -720,20 +720,15 @@ static bool netplay_poll(void)
    if (netplay_is_server(netplay_data) && netplay_data->spectate.enabled)
       return true;
 
-   /* WORKAROUND: The only reason poll_input is ignored in the first frame is
-    * that some cores can't report state size until after the first frame. */
-   if (netplay_data->self_frame_count > 0 || netplay_data->stall || netplay_data->spectate.enabled)
+   /* Read Netplay input, block if we're configured to stall for input every
+    * frame */
+   res = poll_input(netplay_data,
+         (netplay_data->stall_frames == 0)
+         && (netplay_data->read_frame_count <= netplay_data->self_frame_count));
```

Comment:
I'm pretty sure that's a violation of the coding style, but I'm not sure what the proper replacement would be... putting && on the line above, I think?